mlrun/runtimes/nuclio/function.py

```diff
@@ -495,6 +495,11 @@ def add_v3io_stream_trigger(
         extra_attributes = extra_attributes or {}
         if ack_window_size:
             extra_attributes["ackWindowSize"] = ack_window_size
+
+        access_key = kwargs.pop("access_key", None)
+        if not access_key:
+            access_key = self._resolve_v3io_access_key()
```

Comment:
IIUC if the user passes an access key it will still be exposed. can't we mask it?

Reply:
Masking it in the trigger spec is a lot of work.
The access key should be passed in the env var that is already masked, and nuclio will read it from there.

I didn't want to deny the option to pass an access key, but it should be discouraged.

Reply:
If it breaks BC shouldn't we deprecate it first?

Reply:
There is a difference here because it's a serious bug. It shouldn't be considered as breaking BC